[Stella Lye] iP

STELLA-LYE opened the pull request:
### PororoPro

> “Your mind is for having ideas, not holding them.”

 – David Allen ([source](https://dansilvestre.com/productivity-quotes))

PororoPro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use
- All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2324S2/ip/pull/174).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:

- [ ]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Added Level 0

Commit:
Added Level 1

Commit:
Added Level 2

Commit:
Added Level 3

Commit:
Added Level 4

Commit:
Added A-TextUiTesting

Commit:
Level-5

Commit:
Added Level-6

Commit:
Added Level-7

Commit:
Updated Level-7

Commit:
Updated2 Level-7

Commit:
Added Level-8

Commit:
Merge branch 'master' into branch-Level-8

Merge branch-Level-8 with master

Commit:
Added A-MoreOOP

Commit:
Added A-Packages

Commit:
Merge branch 'branch-A-Packages'

A-Packages

Commit:
Merge branch 'master' into add-gradle-support

A-Gradle

Commit:
Added A-JUnit

Commit:
Added A-Jar

Commit:
Added A-JavaDoc

Commit:
Merge branch 'branch-A-JavaDoc'

Added A-JavaDoc

Commit:
Added A-CodingStandard

Commit:
Added Level-9

Commit:
Update A-CodingStandard

Commit:
Merge branch 'branch-A-CodingStandard'

Added A-CodingStandard

Commit:
Resolved merge conflict

Commit:
Update level-9

Commit:
Update ip

Commit:
Add Level-10

Commit:
Merge branch 'branch-Level-10'

Merge Level-10 branch with master

Commit:
Update Level-10 test cases

Commit:
Update and fix bug in Level-10

Commit:
Update with styling

Commit:
Update Level-10 with sendButton functionality

breezetall commented on src/main/java/runner/Parser.java:
```diff
@@ -0,0 +1,178 @@
+                    return delete(arg);
+                default:
+                    return "OOPS!!! I'm sorry, but I don't know what that means :-(";
+            }
```
Should the switch and case be on the same indentation?

breezetall commented on src/main/java/runner/Ui.java:
```diff
@@ -0,0 +1,85 @@
+        return "Noted. I've added this task:\n"
+                + t + "\n"
+                + "Now you have " + n + " tasks in the list.";
+    }
```
Would naming these two methods deleteMsg and addMsg instead be more in line with the coding standard?

breezetall commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,81 @@
+package task;
```
I like how you have implemented most methods in Task for Deadline, Events and Todo. It makes the implementation of the three classes more elegant 👍.

breezetall commented on src/main/java/runner/Parser.java:
```diff
@@ -0,0 +1,178 @@
+    /**
+     * @param info The information given by the user.
+     */
+    public String handle(String info) {
```
Perhaps a more intuitive name like parse?

breezetall commented on src/main/java/runner/Storage.java:
```diff
@@ -0,0 +1,96 @@
+        }
+    }
+
+    /**
+     * Check the status of the Task and add it to the TaskList.
+     * @param line Description of a Task.
+     * @param t Task checked.
+     */
```
Should there be an empty line between description and parameters?

breezetall reviewed (commented):
LGTM. I found your code easy to read and well-written. I noticed just a few minor coding standard violations.

Commit:
Current ip is unable to create JAR files
Edited the build.gradle file to enable JAR file generation
The build.gradle file allows for JAR file generation

Commit:
Current ip has minor method and variable naming issues
Variable and method names are vague and not easy to understand
The variable and method names are edited to be more descriptive and self-explanatory
This way, it will be more readable and understandable to others

Commit:
The exist styling error in the current ip
check_add method name is inconsistent and not descriptive enough
The method name is edited to addCheckedTask
The new method name uses camelCase as suggested and is more readable for others

Commit:
Add A-Assertions

STELLA-LYE committed:
Merge pull request #1 from STELLA-LYE/branch-A-Assertions

Add A-Assertions

Commit:
Add A-CodeQuality

Commit:
Resolve merge conflits

Commit:
Add BCD-Extentions

Commit:
Improve GUI

Commit:
Add unique personality

Commit:
Resolve error

Commit:
Add full GUI window

Commit:
Add a User Guide

Commit:
Add name to titlebar of GUI

Commit:
Update Ui.png

Commit:
Add Pororo.jar file

Commit:
Add jar file

Commit:
Update UserGuide

Commit:
Update jar file
